src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Body,
+  Param,
+  Query,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { AllowedOrganization } from './entities/allowed-organization.entity';
+import { OrganizationStatus } from 'src/common/constants';
+import { GithubInstallationService } from './github.installation.service';
+import { ApiTags } from '@nestjs/swagger';
+import { CreateOrgDto } from './dto/create.org.dto';
+
+@ApiTags('Github-Installation')
+@Controller('github-installation')
+export class GithubInstallationController {
+  constructor(
+    private readonly githubInstallationService: GithubInstallationService,
+  ) {}
+
+  @Post()
+  async create(
+    @Body() createOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.create(createOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
```

Comment:
PullProofAI

Consider consolidating error handling logic to avoid repetition.
```diff
-     try {
-       return this.githubInstallationService.create(createOrgDto);
-     } catch (error) {
-       if (error.status) {
-         throw new HttpException(error.message, error.status);
-       } else {
-         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
-       }
-     }
+     return this.handleServiceError(() => this.githubInstallationService.create(createOrgDto));
```



<!-- This is an auto-generated comment by PullProofAI -->